pkg/build/golang/deps.go

```diff
@@ -0,0 +1,79 @@
+package golang
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/client"
+	"github.com/docker/docker/pkg/archive"
+	"go.uber.org/zap"
+)
+
+func parseDependencies(raw string) map[string]string {
+	rawModules := strings.Split(raw, "\n")
+	modules := map[string]string{}
+
+	for _, module := range rawModules {
+		module = strings.TrimSpace(module)
+		if module == "" {
+			continue
+		}
+
+		parts := strings.Split(module, " ")
+		path := strings.TrimSpace(parts[0])
+		version := ""
+
+		if len(parts) == 2 {
+			version = strings.TrimSpace(parts[1])
+		}
+
+		modules[path] = version
+	}
+
+	return modules
+}
+
+func parseDependenciesFromDocker(ctx context.Context, log *zap.SugaredLogger, cli *client.Client, imageID string) (map[string]string, error) {
+	// Create container
```

Comment:
Nitpick, but these comments are redundant IMO - it is pretty clear from the function call below. Same goes for `Remove container`, and `Copy file from container` and `Unpack file`.